runtime/doc/lsp.txt

```diff
@@ -1306,11 +1307,48 @@ Lua module: vim.lsp.diagnostic                                *lsp-diagnostic*
 
 get_namespace({client_id})                *vim.lsp.diagnostic.get_namespace()*
     Get the diagnostic namespace associated with an LSP client
-    |vim.diagnostic|.
+    |vim.diagnostic| for publishDiagnostics
 
     Parameters: ~
       • {client_id}  (integer) The id of the LSP client
 
+                                     *vim.lsp.diagnostic.get_pull_namespace()*
+get_pull_namespace({client_id})
+    Get the diagnostic namespace associated with an LSP client
+    |vim.diagnostic| for pull diagnostics
```

Comment:
This looks too specialized. Why can't the existing `vim.lsp.diagnostic.get_namespace()` be enhanced ?

Reply:
We could make it accept a flag to choose between the push / pull namespaces? Would that be better?

Reply:
That would be better, but would like to hear @gpanders comments

Reply:
Haven’t had a chance to do thorough code review yet, but why is it not possible to use a single namespace?

Reply:
> but why is it not possible to use a single namespace?

See https://github.com/neovim/neovim/pull/24128#discussion_r1239364111